sympy/sets/sets.py

```diff
@@ -1429,6 +1429,19 @@ def reduce(args):
         if any(s.is_EmptySet for s in args):
             return S.EmptySet
 
+        # If any S.Reals appear, then remove it if there exist a
+        # smaller real subset
+
+        def _fsreal(fs):
+            """Check if all element of FiniteSet is real"""
+            if fs.is_FiniteSet:
+                return all(element.is_real for element in fs)
+
+        if S.Reals in args:
+            other_args = [a for a in args if a != S.Reals]
+            if True in [(b.is_Interval or _fsreal(b)) for b in other_args]:
```

Comment:
You can also use `any` [#1]

[#1]: https://docs.python.org/2/library/functions.html#any